util/kustomize/kustomize.go

```diff
@@ -108,7 +108,9 @@ func (k *kustomize) Build(opts *v1alpha1.ApplicationSourceKustomize, kustomizeOp
 			// set image node:8.15.0 mysql=mariadb alpine@sha256:24a0c4b4a4c0eb97a1aabb8e29f18e917d05abfe1b7a7c07857230879ce7d3d3
 			args := []string{"edit", "set", "image"}
 			for _, image := range opts.Images {
-				args = append(args, string(image))
+				// this allows using ${ARGOCD_APP_REVISION}
+				envSubstitutedImage := envVars.Envsubst(string(image))
```

Comment:
Do we need to be worried about dollar signs in existing image field values? Not sure what's possible in that field.

e.g.

```yaml
- images:
  service=big$money-no-whammies  # I mean this literally, please don't evaluate to `big-no-whammies`.
```

Reply:
Quoting from the docker documentation:
> Name components may contain lowercase letters, digits and separators. A separator is defined as a period, one or two underscores, or one or more dashes. A name component may not start or end with a separator.
> 
> A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes. A tag name may not start with a period or a dash and may contain a maximum of 128 characters.

https://docs.docker.com/engine/reference/commandline/tag/#extended-description

Therefore this case would never happen with a well-formed image name & tag.